bookshelf/api_v3/gce.py

```diff
@@ -0,0 +1,425 @@
+from time import time, sleep
+import uuid
+
+from zope.interface import implementer, provider
+from pyrsistent import PClass, field
+from oauth2client.client import (
+    SignedJwtAssertionCredentials, GoogleCredentials
+)
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+
+from bookshelf.api_v2.logging_helpers import log_green, log_yellow, log_red
+from bookshelf.api_v1 import wait_for_ssh
+from cloud_instance import ICloudInstance, ICloudInstanceFactory, Distribution
+
+
+class GCEConfiguration(PClass):
+    credentials_private_key = field(factory=unicode, initial=u"", mandatory=True)
+    credentials_email = field(factory=unicode, initial=u"", mandatory=True)
+    public_key_filename = field(factory=unicode, mandatory=True)
+    private_key_filename = field(factory=unicode, mandatory=True)
+    project = field(factory=unicode, mandatory=True)
+    machine_type = field(factory=unicode, mandatory=True)
+    image_basename = field(type=unicode, mandatory=True, factory=unicode)
+    username = field(factory=unicode, mandatory=True)
+    description = field(factory=unicode, mandatory=True)
+    instance_name = field(factory=unicode, mandatory=True)
+    base_image_prefix = field(factory=unicode, mandatory=True)
+    base_image_project = field(factory=unicode, mandatory=True)
+
+
+class GCEState(PClass):
+    instance_name = field(factory=unicode, mandatory=True)
+    ip_address = field(factory=unicode, mandatory=True)
+    distro = field(factory=unicode, mandatory=True)
+    zone = field(factory=unicode, mandatory=True)
```

Comment:
We were shooting for configuration (the data in the yaml file) being a static file that doesn't change between provisioning runs.  The region can change based on what you'd like to do and the parameter is set when running `fab up` in on the command line.  Thus, it's not so muh provisioning configuration as a property of the instance you're provisioning.